htdocs/core/lib/company.lib.php

```diff
@@ -390,14 +391,17 @@ function getCountry($searchkey,$withcode='',$dbtouse=0,$outputlangs='',$entconv=
  *    @param	DoliDB		$dbtouse		Database handler (using in global way may fail because of conflicts with some autoload features)
  *    @return   string      				String with state code or state name (Return value is always utf8 encoded and without entities)
  */
-function getState($id,$withcode='',$dbtouse=0)
```

Comment:
To avoid breaking compatiblity, we must not remove a parameter on a function.
Also, getState is a lib function that must be stateless, so behaviour, showing state or not and region or not, is better if it depends on an entry parameter, not on an external setup.
Can you change code so function getState is

function getState($id,$withcode='',$dbtouse=0, $withregion='')

Reply:
Ok, I'm working on it. 
Just a question. What is the purpose of the second parameter on the function call: 
$mysoc->state=getState($mysoc->state_code,0);
Located in commondocgenerator.class.php, Line 108?
$withcode is a string that has no logic built for the numeric value 0.